src/components/AdminPage/UserForm/NewUser.js

```diff
@@ -0,0 +1,67 @@
+import { Row, Col, Button, Container } from "react-bootstrap";
+import styles from "./NewUser.module.css";
+import removeUser from "../../../RestaurantApi/Users/RemoveUser";
+import getAllUsers from "../../../RestaurantApi/Users/GetAllUsers";
+import { useEffect, useState } from "react";
+
+const NewUser = ({ users }) => {
+  const [allUsers, setAllUsers] = useState([]);
+  const [isReloaded, setIsReloaded] = useState({});
+  const removingUser = (id) => {
+    removeUser(id).then(() => setIsReloaded({}));
+  };
+
+  useEffect(() => {
+    const getUsers = async () => {
+      const savedUsers = await getAllUsers();
+      setAllUsers(savedUsers);
+    };
+    const savedUsers = getUsers().catch(
+      (e) => {
+        throw new Error(e.message);
+      },
+      [users, isReloaded]
+    );
+  });
```

Comment:
if there will be enough time,please improve this, because now, useEffect is running in infinity loop, you can check it by doing console.log(savedUsers) in getUsers function. Beside that, getUsers should be outside useEffect I think and this is the function which should be responsible for try and catch